internal/spec/templates/utils.mustache

```diff
@@ -3,317 +3,31 @@ package {{packageName}}
 
 import (
 	"encoding/json"
-	"time"
 )
 
-// PtrBool is a helper routine that returns a pointer to given boolean value.
-func PtrBool(v bool) *bool { return &v }
-
-// PtrInt is a helper routine that returns a pointer to given integer value.
-func PtrInt(v int) *int { return &v }
-
-// PtrInt32 is a helper routine that returns a pointer to given integer value.
-func PtrInt32(v int32) *int32 { return &v }
-
-// PtrInt64 is a helper routine that returns a pointer to given integer value.
-func PtrInt64(v int64) *int64 { return &v }
-
-// PtrFloat32 is a helper routine that returns a pointer to given float value.
-func PtrFloat32(v float32) *float32 { return &v }
-
-// PtrFloat64 is a helper routine that returns a pointer to given float value.
-func PtrFloat64(v float64) *float64 { return &v }
-
-// PtrString is a helper routine that returns a pointer to given string value.
-func PtrString(v string) *string { return &v }
-
-// PtrTime is helper routine that returns a pointer to given Time value.
-func PtrTime(v time.Time) *time.Time { return &v }
-
-type NullableBool struct {
```

Comment:
Interesting that these `Nullable` classes already exist, and appear to do basically the same thing as `Optional`. How come you didn't follow the pattern here and just take advantage of generics to make a `Nullable[T]` type? By following this naming and API, you'd take advantage of existing developer experience with OpenApi-generated Go code.

Reply:
The existing types are kind of similar to what I'm doing here, but not quite. They keep track of set/unset, but don't use that state to control marshaling behavior. Plus, I don't like the name "nullable". It doesn't really communicate the whole purpose. I'm not married to "optional" either, though.